src/core/ext/filters/client_channel/http_proxy.c

```diff
@@ -47,15 +50,18 @@ static char* grpc_get_http_proxy_server(grpc_exec_ctx* exec_ctx) {
     gpr_log(GPR_ERROR, "'%s' scheme not supported in proxy URI", uri->scheme);
     goto done;
   }
-  if (strchr(uri->authority, '@') != NULL) {
-    gpr_log(GPR_ERROR, "userinfo not supported in proxy URI");
-    goto done;
+  char *user_cred_end = strchr(uri->authority, '@');
```

Comment:
For this whole section, you may want to use `gpr_string_split` instead of doing it all manually.